wolfcrypt/src/port/Renesas/renesas_tsip_util.c

```diff
@@ -166,6 +187,333 @@ static byte _tls2tsipdef(byte cipher)
+                                   int countSz, double start, int ret);
+    
+
+    int     ret     = 0; 
+    int     count   = 0;
+    int     times   = 0;
+    double  start;
+    int     numBlocks       = 25; /* how many megs to test (en/de)cryption */
+    word32  bench_size      = (1024ul);
+    int     bench_buf_size  = (int)bench_size;
+    Aes     aes;
+
+    /* key size check. only acceptable 128bit and 256bit key */
+    if( keySz != 16 && keySz != 32)
+        return  1;
+
+
+    if (bench_buf_size % 16)
+        bench_buf_size += 16 - (bench_buf_size % 16);
+    
+    /* buffers passed to TSIP must be 32bit-aligned */  
+    byte* bench_plain  = (byte*)XMALLOC((size_t)bench_buf_size + 16, NULL, DYNAMIC_TYPE_CIPHER);
+    byte* bench_cipher = (byte*)XMALLOC((size_t)bench_buf_size + 16, NULL, DYNAMIC_TYPE_CIPHER);
+
+
+
+    XMEMSET(bench_plain,    0x30, bench_buf_size +16);
+    XMEMSET(bench_cipher,   0,    bench_buf_size +16);  
+    XMEMSET(&aes,           0,    sizeof(aes));
+    
+
+    /* 
+     * The key passed as a parameter of this function should be "installed"
+     * into TSIP with a provisioning key. To avoid this troublesom,   
+     * make TSIP generate AES key in stead and use it for encryption and
+     * decryption.
+     */
+    aes.ctx.keySize = keySz;
+
+    XMEMCPY((byte*)aes.reg, iv,ivSz );  /* copy IV */    
+
+    if(keySz == 16){
+        ret = R_TSIP_GenerateAes128RandomKeyIndex(&aes.ctx.tsip_keyIdx);
+    }
+    else{
+        ret = R_TSIP_GenerateAes256RandomKeyIndex(&aes.ctx.tsip_keyIdx);
+    }
+
+    if( ret!= TSIP_SUCCESS)
+    {
+        WOLFSSL_MSG("R_TSIP_GenerateAesXXXRandomKeyIndex failed" );
+        ret = -1;
+        goto exit;
+    }
+  
+    bench_stats_start(&count, &start);
+
+    do {
+        for (times = 0; times < numBlocks; times++ ) {
+
+            ret = wc_tsip_AesCbcEncrypt(
+                    &aes,            
+                    bench_cipher,
+                    bench_plain,
+                    bench_size);
+    
+            if( ret != 0 ){
+                WOLFSSL_MSG("wc_tsip_AesCbcEncrypt failed" );
+                goto exit;
+            }
+        }
+        count += times;
+    } while (bench_stats_sym_check(start));
+   
+    bench_stats_sym_finish(encLabel, doAsync, count, bench_size, start, ret);
+
+
+    bench_stats_start(&count, &start);
+
+    do {
+        for (times = 0; times < numBlocks ; times++) {
+ 
+            ret = wc_tsip_AesCbcDecrypt(
+                    &aes,               
+                    bench_plain,
+                    bench_cipher,
+                    bench_size);
+            if( ret != 0){
+                WOLFSSL_MSG("wc_tsip_AesCbcDecrypt failed");
+                goto exit;
+            }
+        }
+        count += times;
+    } while (bench_stats_sym_check(start));
+
+    bench_stats_sym_finish(decLabel, doAsync, count, bench_size, start, ret);
+
+
+    (void)decLabel;
+
+exit:
```

Comment:
See note above and coding standards about use of goto

Reply:
Rewrote code without "goto".